packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
```

Comment:
This does use `testWidgets` just indirectly through `asyncTest` - the fact that it's centralized actually makes it easier to ensure consistent use throughout.

The files won't always be individually runnable, especially when they use API since there's no way to reset the native code in between tests. I've opted to not add 5+ files for these changes though, since they will be individually runnable when API dart is merged.

This is a stacked PR against #2240 so any tests here will need those changes to run.